### PR TITLE
Jinja templater: Allow "load_macros_from_path" to be a comma-separated list of paths

### LINKED ISSUE #829: Are there plans to support Jinja `include` and `from` statements?
Currently, when trying to lint an sql file with one of these jinja statements:

```jinja
{% from 'my_macros.sql' import a_macro %}
```
or 
```jinja
{% include 'file.sql' %}
```
it will fail with the message: "no loader for this environment specified".

Unless I'm missing something, this seems to be caused by the use of a SandboxedEnvironment with, as the message says, no loader to load either _my_macros.sql_ or _file.sql_.

Are there any plans to support these? Or maybe I'm doing something wrong?

Thank you.

### PR DESCRIPTION
**Pull request by barrywhart**
### Brief summary of the change made
Fixes #829

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/core/templaters/jinja.py [hunk 1]
```diff
@@ -212,13 +214,16 @@ def _get_jinja_env(self, config=None):
...
-    def _get_macros_path(self, config):
-        macros_path = None
+    def _get_macros_path(self, config: FluffConfig) -> Optional[List[str]]:
         if config:
             macros_path = config.get_section(
                 (self.templater_selector, self.name, "load_macros_from_path")
             )
-        return macros_path
+            if macros_path:
+                result = [s.strip() for s in macros_path.split(",") if s.strip()]  # type: ignore
```
Comment (jpy-git) [inline src/sqlfluff/core/templaters/jinja.py:223]: This mypy error is due to some dodgy typing in `FluffConfig.get_section`:
image

That return type should really just be `Any` so could you update that and remove the type ignore here

**Review by jpy-git — COMMENTED** [review]
Looks good, couple small points

File: src/sqlfluff/core/templaters/jinja.py [hunk 2]
```diff
@@ -64,31 +65,32 @@ def _extract_macros_from_template(template, env, ctx):
...
-            for dirpath, _, files in os.walk(path):
-                for fname in files:
-                    if fname.endswith(".sql"):
-                        macro_ctx.update(
-                            cls._extract_macros_from_path(
-                                os.path.join(dirpath, fname), env=env, ctx=ctx
+        for path_entry in path:
+            # Does it exist? It should as this check was done on config load.
+            if not os.path.exists(path_entry):  # pragma: no cover
+                raise ValueError(f"Path does not exist: {path}")
```
[the remark this answers is not in the document]
Comment (barrywhart) [reply src/sqlfluff/core/templaters/jinja.py]: Done! See `test__templater_jinja_error_macro_path_does_not_exist`.

File: src/sqlfluff/core/templaters/jinja.py (hunk 1 above)
Comment (barrywhart) [reply src/sqlfluff/core/templaters/jinja.py]: Done!

File: src/sqlfluff/core/templaters/jinja.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpy-git) [reply src/sqlfluff/core/templaters/jinja.py]: Thanks! 🥇

**Review by jpy-git — APPROVED** [review]
@barrywhart All looks good to me, nice work! 🎉 LGTM 🚀 ✨

**Pull request merged**